src/audio/mixin_mixout.c

```diff
@@ -124,12 +129,50 @@ struct mixout_data {
 	struct mixed_data_info *mixed_data_info;
 };
 
+/* large_config_set param id for mixer mode config */
+#define MIXER_MODE_PARAM_ID 1
+
+struct mixer_mode_sink_config {
+	/* Index of output queue (aka sink) this config is for,
+	 * range from 0 to MIXIN_MAX_SINKS - 1
+	 */
+	uint32_t output_queue_id;
+
+	/* Channel remapping config is here which is not (yet) supported. */
+	uint32_t reserved1;
+	uint32_t reserved2;
+	uint32_t reserved3;
+
+	/* Gain to be applied to input signal. Valid range: 0x0..0x400 (0.0 <= gain <= 1.0). Values
+	 * greater than 0x400 are treated as 0x400 (unity gain). To apply gain, multiply sample
+	 * by "gain" and divide by 1024.
+	 */
+	uint16_t gain;
+	uint16_t reserved4;
+} __packed __aligned(4);
```

Comment:
Should these packed structures be in an IPC header ?

Reply:
You are right. Will move then there.

Reply:
Moved them to include/ipc4/mixin_mixout.h